hieradocs.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+#  hiera::key::name:
+#    datatype: String.
+#    description: |
+#      String describing the key.
+#    found_in:
+#    - someFile.yaml
+
+SAVEIFS=$IFS
+IFS=$(echo -en "\n\b")
+
+PATTERN_PARAM=\[[*]\([^*]*\)[*]\]
+PATTERN_COMMENT=\(.+\)
+
+HIERA_DIR=../hieradata
```

Comment:
I forgot that I hard coded that.  I think this is OK for now but maybe an issue for that dir to be taken as a command line argument would be good.